health-services/resource-estimation-service/src/main/java/org/egov/processor/util/CalculationUtil.java

```diff
@@ -0,0 +1,132 @@
+import java.util.Map;
+
+import org.egov.processor.config.ServiceConstants;
+import org.egov.processor.web.models.Assumption;
+import org.egov.processor.web.models.Operation;
+import org.egov.processor.web.models.PlanConfiguration;
+import org.egov.processor.web.models.PlanConfigurationRequest;
+import org.egov.tracer.model.CustomException;
+import org.springframework.stereotype.Component;
+
+import static org.egov.processor.config.ServiceConstants.PROPERTIES;
+
+@Component
+public class CalculationUtil {
+	
+	private PlanUtil planUtil;
+	
+	
+
+    public CalculationUtil(PlanUtil planUtil) {
+		this.planUtil = planUtil;
+	}
+
+    /**
+     * Calculates the output value based on the input value, operator, and assumption value.
+     *
+     * @param input          The input value.
+     * @param operator       The operator enum.
+     * @param assumptionValue The assumption value.
+     * @return The calculated output.
+     */
+    public BigDecimal calculateOutputValue(BigDecimal input, Operation.OperatorEnum operator, BigDecimal assumptionValue) {
+        return switch (operator) {
+            case PLUS -> input.add(assumptionValue);
+            case MINUS -> input.subtract(assumptionValue);
+            case SLASH -> input.divide(assumptionValue,2,RoundingMode.DOWN).setScale(2);
+            case STAR -> input.multiply(assumptionValue);
+            case PERCENT -> input.remainder(assumptionValue);
+            case _U -> input.pow(assumptionValue.intValue());
+            default -> throw new CustomException("UNSUPPORTED_OPERATOR", "Unsupported operator: " + operator);
+        };
+    }
+
+    /**
+     * Converts a list of assumptions into a map with assumption keys as keys and assumption values as values.
+     *
+     * @param assumptions The list of assumptions to convert.
+     * @return The map of assumptions.
+     */
+    public Map<String, BigDecimal> convertAssumptionsToMap(List<Assumption> assumptions) {
+        Map<String, BigDecimal> assumptionMap = new HashMap<>();
+        for (Assumption assumption : assumptions) {
+            assumptionMap.put(assumption.getKey(), assumption.getValue());
+        }
+        return assumptionMap;
+    }
+
+    /**
+     * Calculates resources based on the provided JSON node, list of operations, and assumption values.
+     *
+     * @param jsonNode           The JSON node containing the data.
+     * @param resultMap          The map to store the results.
+     * @param mappedValues       The mapped values for inputs.
+     * @param assumptionValueMap The assumption values map.
+     */
+    public void calculateResources(JsonNode jsonNode, PlanConfigurationRequest planConfigurationRequest, Map<String, BigDecimal> resultMap,
+                                Map<String, String> mappedValues, Map<String, BigDecimal> assumptionValueMap) {
+    	PlanConfiguration planConfig = planConfigurationRequest.getPlanConfiguration();
+        for (JsonNode feature : jsonNode.get("features")) {
```

Comment:
Why is this a nested loop, explain the logic for nesting. is it not possible to simplify ?

Reply:
This method will calculate resource based on operation per row hence it required loop. It will also create a micro plan row wise by putting it on topic which will be consumed by plan service.